addons/hr_holidays/report/holidays_summary_report.py

```diff
@@ -73,36 +84,66 @@ def _get_leaves_summary(self, start_date, empid, holiday_type):
                 if date_from >= start_date and date_from <= end_date:
-                    res[(date_from-start_date).days]['color'] = holiday.holiday_status_id.color_name
+                    res[(date_from-start_date).days]['color'] = COLORS_MAP[holiday.holiday_status_id.color]
                 date_from += timedelta(1)
             count += holiday.number_of_days
         employee = self.env['hr.employee'].browse(empid)
         return {'emp': employee.name, 'display': res, 'sum': count}
 
+    def _get_employees(self, data):
+        if 'depts' in data:
+            return self.env['hr.employee'].search([('department_id', 'in', data['depts'])])
+        elif 'emp' in data:
+            return self.env['hr.employee'].browse(data['emp'])
+        return self.env['hr.employee'].search([])
+
     def _get_data_from_report(self, data):
         res = []
-        Employee = self.env['hr.employee']
         if 'depts' in data:
-            for department in self.env['hr.department'].browse(data['depts']):
+            employees = self._get_employees(data)
+            departments = self.env['hr.department'].browse(data['depts'])
+            for department in departments:
                 res.append({
                     'dept': department.name,
                     'data': [
                         self._get_leaves_summary(data['date_from'], emp.id, data['holiday_type'])
-                        for emp in Employee.search([('department_id', '=', department.id)])
+                        for emp in employees.filtered(lambda emp: emp.department_id.id == department.id)
                     ],
                     'color': self._get_day(data['date_from']),
                 })
         elif 'emp' in data:
             res.append({'data': [
                 self._get_leaves_summary(data['date_from'], emp.id, data['holiday_type'])
-                for emp in Employee.browse(data['emp'])
+                for emp in self._get_employees(data)
             ]})
         return res
 
-    def _get_holidays_status(self):
+    def _get_leaves(self, date_from, employees, holiday_type, date_to=None):
+        state = ['confirm', 'validate'] if holiday_type == 'both' else ['confirm'] if holiday_type == 'Confirmed' else ['validate']
+
+        if not date_to:
+            date_to = date_from + relativedelta(days=59)
+
+        return self.env['hr.leave'].search([
+            ('employee_id', 'in', employees.ids),
+            ('state', 'in', state),
+            ('date_from', '<=', str(date_to)),
+            ('date_to', '>=', str(date_from))
+        ])
+
+    def _get_holidays_status(self, data):
         res = []
-        for holiday in self.env['hr.leave.type'].search([]):
-            res.append({'color': holiday.color_name, 'name': holiday.name})
+        employees = self.env['hr.employee']
+        if 'depts' in data:
+            employees = self._get_employees(data)
+        elif 'emp' in data:
+            employees = self._get_employees(data)
```

Comment:
Could be simplified a bit:
```suggestion
        if 'depts' in data or 'emp' in data:
            employees = self._get_employees(data)
```
or 
```suggestion
        if {'depts', 'emp'} & data.keys():
            employees = self._get_employees(data)
```

Reply:
My bad, thanks!